src/Protobuf/Decode.elm

```diff
@@ -726,16 +780,39 @@ tagDecoder =
             )
 
 
-varIntDecoder : Decode.Decoder ( Int, Int )
-varIntDecoder =
+intDecoder : IntOperations int -> Decoder int
+intDecoder =
+    varIntDecoder >> packedDecoder VarInt
+
+
+sintDecoder : IntOperations int -> Decoder int
+sintDecoder config =
+    varIntDecoder config
+        |> Decode.map (Tuple.mapSecond config.fromZigZag)
+        |> packedDecoder VarInt
```

Comment:
So in decoding we use `fromZigZag` and `toUnsigned`. Although this is the correct thing to do, it confuses me. We should probably think of a better name for the unsigned methods.

Reply:
I guess the thing is we're not converting between signed/unsigned. We're just correcting for the fact we're nog using 32 bit integers.

Reply:
Maybe just `fromSigned` and `toSigned`? 🙂

Reply:
Good idea, done